### PR TITLE
implement unlock & training modal flow

### PR DESCRIPTION
**Pull request by towardsnewtech**

### CODE REVIEW

File: components/ProfilePage/Desktop/PreReveal/PreProflileInformation.tsx [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   const { userInfo } = useUserProvider()
+  const {
+    isEditable,
+    editedBio,
+    editedINeedHelpWith,
+    editedAskedMeAbout,
+    handleEditedAskedMeAbout,
+    handleEditedINeedHelpWith,
+    handleEditedBio,
+  } = useProfileProvider()
```
Comment (sweetmantech) [inline components/ProfilePage/Desktop/PreReveal/PreProflileInformation.tsx]: 🧠 💡 👍

File: providers/ProfileContext.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      setEditedLocation(userInfo.location)
+      setEditedAskedMeAbout(userInfo.askMeAbout)
+      setEditedINeedHelpWith(userInfo.iNeedHelpWith)
+      setEditedBio(userInfo.bio)
+    }
+  }, [isEditable, userInfo])
+
+  const handleEditable = (ediable: boolean) => {
+    setIsEditable(ediable)
+  }
```
Comment (sweetmantech) [inline providers/ProfileContext.tsx]: why are we exporting  a wrapper`handleEditable`?

why not just export the base `setIsEditable`?

the naming feels more clear to me with `setIsEditable` 👍  while `handleEditable` feels confusing as to what the function does. 💭

File: providers/ProfileContext.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    setEditedBio(e.target.value)
+  }
+
+  const handleEditedLocation = (e: any) => {
+    setEditedLocation(e.target.value)
+  }
+
+  const handleExpandedMore = (isExpanded: boolean) => {
+    setExpandedMore(isExpanded)
+  }
```
Comment (sweetmantech) [inline providers/ProfileContext.tsx]: why are we wrapping all these state setters?

no additional logic, right?

why not just export the setters directly?

File: components/ProfilePage/PreDesktopProfileView.tsx [hunk 4]
```diff
@@ -133,19 +108,11 @@ const PreDesktopProfileView: FC<ProfileViewProps> = ({
...
           <div className="flex flex-col gap-y-[40px]">
-            <PreProfileInformation
-              editedAskedMeAbout={editedAskedMeAbout}
-              editedINeedHelpWith={editedINeedHelpWith}
-              editedBio={editedBio}
-              handleEditedAskedMeAbout={handleEditedAskedMeAbout}
-              handleINeedHelpWith={handleINeedHelpWith}
-              handleEditedBio={handleEditedBio}
-              isEditable={isEditable}
-            />
```
Comment (sweetmantech) [inline components/ProfilePage/PreDesktopProfileView.tsx:144]: waaay cleaner with provider, nice work ✨

File: components/ProfilePage/ProfilePage.tsx [hunk 5]
```diff
@@ -127,48 +53,10 @@ const ProfilePage = () => {
...
-            handleEditedTwitterHandle={handleEditedTwitterHandle}
-            handleEditedLocation={handleEditedLocation}
-            isEditable={isEditable}
-            editedAskedMeAbout={editedAskedMeAbout}
-            editedUserName={editedUserName}
-            editedLocation={editedLocation}
-            editedINeedHelpWith={editedINeedHelpWith}
-            editedTwitterHandle={editedTwitterHandle}
-            editedBio={editedBio}
-          />
```
Comment (sweetmantech) [inline components/ProfilePage/ProfilePage.tsx:171]: makes me happy to see these components getting smaller 😊

File: pages/profile/[address].tsx [hunk 6]
```diff
@@ -48,6 +52,22 @@ export const getServerSideProps: GetServerSideProps<ProfileProps> = async ({ par
...
+  useEffect(() => {
+    getUserData(router.query.address as string)
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router])
+
+  return (
+    <ProfileProvider>
+      <ProfilePage />
+    </ProfileProvider>
```
Comment (sweetmantech) [inline pages/profile/[address].tsx]: PERFECT wrapping of Provider only around specific components 👏

**Review by SweetmanTech — APPROVED** [review]
ship it 🚢

**Review by SweetmanTech — APPROVED** [review]
ship it 🚢

**Pull request merged**